build/dsc/resource.rb

```diff
@@ -93,26 +93,29 @@ def absentable?
       @absentable
     end
 
-    def absent_value
-      @absent_value ||=
-        if ensure_property.values.any? { |v| v.casecmp('absent') == 0 }
-          'absent'
-        elsif ensure_property.values.any? { |v| v.casecmp('disable') == 0 }
-          'disable'
-        else
-          throw 'Error processing MOF schema - could not determine equivalent \'absent\' value for ensure'
-        end
+    def ensure_hash
```

Comment:
nit - this returns an array, but states that it's a hash.  I think it might be better to index by name like:

```ruby
{
  'MSFT_xRunbookDirectory' => { :present => ['published', 'draft'], :absent => [ 'absent'] },
  'MSFT_WindowsOptionalFeature' => { :present => ['enable'], :absent => [ 'disable'] }
}
```